src/reports.coffee

```diff
@@ -83,7 +83,10 @@ sendReports = (job, flag, done) ->
 
               # If report has been fetched get it from the map
               if channelReportMap[channel._id]
-                reportMap[userKey].data.push channelReportMap[channel._id]
+                data = channelReportMap[channel._id]
+                # add report - always add if the channel is enabled, otherwise only if there is data
+                if data.channel.status is 'enabled' or data.data.length isnt 0
```

Comment:
Is channel status always set? Just thinking of channels that were have never had their status explicitly set (they existed before the functionality was added to core). Can status ever be empty?